src/web/app/pages-instructor/instructor-session-individual-extension-page/individual-extension-date-modal/individual-extension-date-modal.component.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import SpyInstance = jest.SpyInstance;
+import { SimpleModalService } from '../../../../services/simple-modal.service';
+import { TimezoneService } from '../../../../services/timezone.service';
+import { createMockNgbModalRef } from '../../../../test-helpers/mock-ngb-modal-ref';
+import {
+  FeedbackSession,
+  FeedbackSessionPublishStatus, FeedbackSessionSubmissionStatus, ResponseVisibleSetting, SessionVisibleSetting,
+} from '../../../../types/api-output';
+import { InstructorSessionIndividualExtensionPageModule } from '../instructor-session-individual-extension-page.module';
+import { IndividualExtensionDateModalComponent, RadioOptions } from './individual-extension-date-modal.component';
+
+describe('IndividualExtensionDateModalComponent', () => {
+    const testFeedbackSession: FeedbackSession = {
+      courseId: 'testId1',
+      timeZone: 'Asia/Singapore',
+      feedbackSessionName: 'Test Session',
+      instructions: 'Instructions',
+      submissionStartTimestamp: 1000000000000,
+      submissionEndTimestamp: 1500000000000,
+      gracePeriod: 0,
+      sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+      responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
+      submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+      publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
+      isClosingEmailEnabled: true,
+      isPublishedEmailEnabled: true,
+      createdAtTimestamp: 0,
+      studentDeadlines: {},
+      instructorDeadlines: {},
+    };
+
+    const testTimeString = '5 Apr 2000 2:00:00';
+
+    let component: IndividualExtensionDateModalComponent;
+    let fixture: ComponentFixture<IndividualExtensionDateModalComponent>;
+    let simpleModalService: SimpleModalService;
+    let timeZoneService: TimezoneService;
+
+    beforeEach(waitForAsync(() => {
+        TestBed.configureTestingModule({
+          imports: [
+            HttpClientTestingModule,
+            InstructorSessionIndividualExtensionPageModule,
+          ],
+          providers: [NgbActiveModal],
+        })
+        .compileComponents();
+      }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(IndividualExtensionDateModalComponent);
+        component = fixture.componentInstance;
+        simpleModalService = TestBed.inject(SimpleModalService);
+        timeZoneService = TestBed.inject(TimezoneService);
+        jest.spyOn(timeZoneService, 'formatToString').mockReturnValue(testTimeString);
+        component.numStudents = 10;
+        component.numInstructors = 20;
+        fixture.detectChanges();
+      });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+      });
+
+    it('should snap with default fields', () => {
+        expect(component).toBeTruthy();
+    });
```

Comment:
Test does not match description, this should be a snapshot test with default values.